Fix autoskipping closing chevrons in rust-mode

fmdkdd opened the pull request:
Before this commit, closing chevrons were not autoskipped.  Typing:

`T<` would result in the correct insertion of a matching closing
chevron: `T<|>`, but typing `>` would not result in `T<>|>`, not
skipping the closing chevron even with the `autoskip` action set on the
pair.

This was caused by the regexp in `sp-rust-could-be-parameterized` set as
a `when` filter for the local pair "<" ">".  This filter is called by
`sp--do-action-p` not only when inserting a chevron, but also when
determining if the closing chevron should be autoskipped.  In the
closing case, the regexp does not match, because at the moment this
check is done the code contains one extra chevron: `<T>|>`.

This commits adds a ">" alternative to the regexp so that filter returns
true when `sp--do-action-p` is called for the autoskip action on the
closing chevron.

Closes #642.

fmdkdd commented:
@Fuco1 It's fixed for me.  But I think I want to add a test to ensure this stays fixed in the future.

fmdkdd commented:
I've added a test for this particular case and amended the commit.  Should be good to go.

Fuco1 commented:
Another solution would be check the `action` parameter for `'insert` and only do the check then.  Is the added code useful in all the variants?

In the new system you will have an option to add predicates for each action separately which will really help the reuse... now you need to switch/case/cond each one which is quite ugly.

Anyway I'm happy with any fix :)

fmdkdd commented:
@Fuco1 Right!  I rewrote the filter to be more explicit.  It still passes all the tests, and I've added a couple to catch this particular situation with autoskipping.

Let me know what you think.

Fuco1 commented:
Great, I have nothing to add :) Thank you.

fmdkdd commented:
Thank you for your help :)
